prog.go

```diff
@@ -155,14 +155,32 @@ func (p *BPFProg) SetAttachTarget(attachProgFD int, attachFuncName string) error
 	return nil
 }
 
-// TODO: fix API to return error
+// Deprecated: use BPFProg.SetExpectedProgramType() instead.
 func (p *BPFProg) SetProgramType(progType BPFProgType) {
-	C.bpf_program__set_type(p.prog, C.enum_bpf_prog_type(int(progType)))
+	C.bpf_program__set_type(p.prog, C.enum_bpf_prog_type(C.int(progType)))
 }
 
-// TODO: fix API to return error
+func (p *BPFProg) SetExpectedProgramType(progType BPFProgType) error {
+	retC := C.bpf_program__set_type(p.prog, C.enum_bpf_prog_type(C.int(progType)))
```

Comment:
Have you tried as?

```suggestion
	retC := C.bpf_program__set_type(p.prog, C.enum_bpf_prog_type(progType))
```

If a no go, use only:

```suggestion
	retC := C.bpf_program__set_type(p.prog, C.int(progType))
```